test/ui/07 - feedback/feedback.spec.js

```diff
@@ -0,0 +1,192 @@
+        });
+
+        it('should not allow them to send feedback with no rating', async() => {
+          await page.fill('#feedbackText', config.validQuery);
+          await page.fill('#feedbackEmail', config.validEmail);
+          await submitPage();
+
+          const errorSummaries = await getErrorSummaries();
+          const errorMessages = await getErrorMessages();
+
+          const expected = 'Select how you feel about the service you received';
+
+          expect(await page.url()).to.equal(baseURL + '/feedback');
+          expect(errorSummaries.length).to.equal(1);
+          expect(await errorSummaries[0].innerText()).to.equal(expected);
+          expect(errorMessages.length).to.equal(1);
+          expect(await errorMessages[0].innerText()).to.equal(expected);
+        });
+
+        it('should not allow them to send feedback with no improvements', async() => {
+          const radios = await page.$$('input[type="radio"]');
+          await radios[0].click();
+          await page.fill('#feedbackEmail', config.validEmail);
+          await submitPage();
+
+          const errorSummaries = await getErrorSummaries();
+          const errorMessages = await getErrorMessages();
+
+          const expected = 'Enter feedback';
+
+          expect(await page.url()).to.equal(baseURL + '/feedback');
+          expect(errorSummaries.length).to.equal(1);
+          expect(await errorSummaries[0].innerText()).to.equal(expected);
+          expect(errorMessages.length).to.equal(1);
+          expect(await errorMessages[0].innerText()).to.equal(expected);
+        });
+
+        it('should not allow them to send feedback with an invalid email', async() => {
+          const radios = await page.$$('input[type="radio"]');
+          await radios[0].click();
+          await page.fill('#feedbackText', config.validQuery);
+          await page.fill('#feedbackEmail', config.invalidEmail);
+          await submitPage();
+
+          const errorSummaries = await getErrorSummaries();
+          const errorMessages = await getErrorMessages();
+
+          const expected = 'Enter a valid email address';
+
+          expect(await page.url()).to.equal(baseURL + '/feedback');
+          expect(errorSummaries.length).to.equal(1);
+          expect(await errorSummaries[0].innerText()).to.equal(expected);
+          expect(errorMessages.length).to.equal(1);
+          expect(await errorMessages[0].innerText()).to.equal(expected);
+        });
+
+        it('should allow them to send feedback with no email', async() => {
+          const radios = await page.$$('input[type="radio"]');
+          await radios[0].click();
+          await page.fill('#feedbackText', config.validQuery);
+          await submitPage();
+          // valid so page will be submitted page
+          expect(await page.url()).to.equal(baseURL + '/feedback-submitted');
+        });
+
+      });
+
+    });
+
+    describe('Sending feedback', () => {
```

Comment:
Update to be more descriptive of the user's action:

`when the user submits the feedback page with valid rating and text`